src/main/java/fi/aalto/cs/apluscourses/model/Points.java

```diff
@@ -103,4 +123,19 @@ private static void parseSubmissions(@NotNull JSONObject exerciseJson,
     submissions.put(exerciseId, submissionIds);
   }
 
+  @Nullable
+  private static Long parseSubmissionId(String submissionUrl) {
+    if (submissionUrl.isEmpty()) {
+      return null;
+    }
+    if (submissionUrl.endsWith("/")) {
+      submissionUrl = submissionUrl.substring(0, submissionUrl.length() - 1);
+    }
```

Comment:
This logic could be put in a separate utility function `StringUtil.removeTrailingSubstring(str, substr)`.